json/writer.go

```diff
@@ -1,32 +1,16 @@
 // SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 
-package spdx_json
+package json
 
 import (
 	"encoding/json"
-	"github.com/spdx/tools-golang/spdx/v2_3"
 	"io"
 
-	"github.com/spdx/tools-golang/spdx/v2_2"
+	"github.com/spdx/tools-golang/spdx/common"
 )
 
-// Save2_2 takes an SPDX Document (version 2.2) and an io.Writer, and writes the document to the writer in JSON format.
-func Save2_2(doc *v2_2.Document, w io.Writer) error {
-	buf, err := json.Marshal(doc)
-	if err != nil {
-		return err
-	}
-
-	_, err = w.Write(buf)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// Save2_3 takes an SPDX Document (version 2.2) and an io.Writer, and writes the document to the writer in JSON format.
-func Save2_3(doc *v2_3.Document, w io.Writer) error {
+// Write takes an SPDX Document and an io.Writer, and writes the document to the writer in JSON format.
+func Write(doc common.Document, w io.Writer) error {
```

Comment:
Curious about the chain converter you've implemented, what would happen if we created another chain for backwards conversion and have an option to say what version it would like to be exported as?

Reply:
It already supports both forward and backwards conversion!

Let's say you have the latest model and want to export v2.2, the code is something like this:

```go
// doc is an spdx.Document from somewhere, or it could be a different version, like v2_2.Document etc.
doc := spdx.Document{}
// make the result document version we want:
out := v2_2.Document{}
// call convert, which handles the backwards migration as well:
convert.DocumentChain.Convert(doc, &out)
```